features/testbot/trip.feature

```diff
@@ -48,12 +47,130 @@ Feature: Basic trip planning
             | kl    |
             | la    |
 
-
         When I plan a trip I should get
             | waypoints               | trips         |
             | a,b,c,d,e,f,g,h,i,j,k,l | cbalkjihgfedc |
 
-    Scenario: Testbot - Trip planning with multiple scc
+
+    Scenario: Testbot - Trip planning with less than ten node tfse
+        Given the query options
+            | source        | 0       |
+            | destination   | 2       |
+
+        Given the node map
+            """
+            a  b
+
+                  c
+            e  d
+            """
+ 
+         And the ways
+            | nodes |
+            | ab    |
+            | ac    |
+            | ad    |
+            | ae    |
+            | bc    |
+            | bd    |
+            | be    |
+            | cd    |
+            | ce    |
+            | de    |
+ 
+         When I plan a trip I should get
+            |  waypoints  | trips  | durations         | distance |
+            |  a,b,c,d,e  | abedc  | 8.200000000000001 | 81.6     |
+
+
+    Scenario: Testbot - Trip planning with more than 10 nodes tfse
+        Given the query options
+            | source        | 0       |
+            | destination   | 10      |
+
+        Given the node map
+            """
+            a b c d e f g h i j k
+            """
+
+        And the ways
+            | nodes |
+            | ab    |
+            | bc    |
+            | cd    |
+            | de    |
+            | ef    |
+            | fg    |
+            | gh    |
+            | hi    |
+            | ij    |
+            | jk    |
+
+        When I plan a trip I should get
+            |  waypoints              | trips       | durations  | distance  |
+            |  a,b,c,d,e,f,g,h,i,j,k  | abcdefghijk | 10         | 99.9      |
+
+
+    Scenario: Testbot - Trip planning with illegal destination
+        Given the query options
+            | source        | 2       |
+            | destination   | 10      |
+
+        Given the node map
+            """
+            a b c
+            """
+
+        And the ways
+            | nodes |
+            | ab    |
+            | bc    |
+
+        When I plan a trip I should get
+            |  waypoints  | status        | message                                                                                  |
+            |  a,b,c      | InvalidInputs | Source or destination indices are greater number of coordinates provided. |
+
+
+    Scenario: Testbot - Trip planning with illegal source
+        Given the query options
+            | source        | 10     |
```

Comment:
`source` and `destination` should be columns in the "When I plan a trip I should get" columns, not a separate "Given", allowing per-query adjustment.  This will allow some consolidation of these test cases.